File: Programming and Projection Methods/Lab2/Tema2/src/main/java/domain/Booking.java
package domain;

import javafx.util.Pair;
import repository.HasID;

public class Booking implements HasID<Pair<Client,Ride>> {
    private Client client;
    private Ride ride;
    private int nr_places_wanted;
    private String places;

    public Client getClient() {
        return client;
    }

    public Ride getRide() {
        return ride;
    }

    public Booking(Client client, Ride ride, int places_wanted, String places){
        this.client=client;
        this.ride=ride;
        this.nr_places_wanted=places_wanted;
        this.places=places;
    }

    @Override
    public Pair<Client,Ride> getID() {
        return new Pair<>(client,ride);
    }

    @Override
    public void setID(Pair<Client, Ride> clientRidePair) {
        this.client=clientRidePair.getKey();
        this.ride=clientRidePair.getValue();
    }

    public int getNr_places_wanted(){
        return nr_places_wanted;
    }

    public void setNr_places_wanted(int nr){
        this.nr_places_wanted=nr;
    }

    public String getPlaces() {
        return places;
    }

    public void setPlaces(String places) {
        this.places = places;
    }
}

File: Programming and Projection Methods/Lab5C#/Lab5/Lab5/domain/Booking.cs
using Lab5.repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab5.domain
{
    public class Booking:IHasID<KeyValuePair<Ride, Client>>
    {
        public Booking(KeyValuePair<Ride, Client> id, int nrPlacesWanted, string places)
        {
            Id = id;
            NrPlacesWanted = nrPlacesWanted;
            Places = places;
        }

        public KeyValuePair<Ride,Client> Id { get; set; }
        public int NrPlacesWanted { get; set; }
        public string Places { get; set; }


    }
}
